### ISSUE TITLE
Incorrect Placeholder Check for Plural Forms in Weblate

### ISSUE THREAD

**Issue by eschumacher-99** [WeblateOrg/weblate#13016]
### Describe the issue

I encountered a bug in Weblate related to the placeholder check when handling plural forms. Specifically, when plural forms use variables that differ from those in the singular form, Weblate incorrectly validates the placeholders in the translation.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar filed issues in this repository.

### Steps to reproduce the behavior

1. Create a translation entry with the following:
Singular form: @(count) Stck.
Plural form: @(otherVariable) Stck.

2. Use the placeholder check feature in Weblate: 
placeholders:r"@\([^)]+\)"

3. Attempt to translate both forms.
4. The placeholder check incorrectly validates the plural form.
Weblate expects the variable @(count) in the plural form, marking @(otherVariable) as superfluous and @(count) as missing.

### Expected behavior

The placeholder check should correctly recognize and validate the variables specific to each form (plural or singular). In this case, it should acknowledge @(otherVariable) as the appropriate variable for the plural form without expecting @(count).

### Screenshots

![image](https://github.com/user-attachments/assets/eed8c086-d9af-4e99-a979-090a0f3e72e7)

### How do you run Weblate?

PyPI module

### Weblate versions

5.4.2

**Comment by nijel** [comment]
Indeed, this situation is not gracefully handled. On the other side I think this is a bad approach because it will become tricky once the target language has a different number of plurals. The `PlaceholderCheck` could be updated to use `PluralMapper`, but still, this can't handle 100% of cases with different number of plurals.
